libfuse/special_read_file.go

```diff
@@ -16,26 +17,51 @@ import (
 // a function.
 type SpecialReadFile struct {
 	read func(context.Context) ([]byte, time.Time, error)
+
+	dataLock   sync.Mutex
```

Comment:
There's an equivalent type in libdokan, too. Perhaps fix that one also? Up to you.